addons/rookupgrade/10to14/install.sh

```diff
@@ -25,148 +29,222 @@ function rookupgrade_10to14_upgrade() {
+                    bail "Failed to verify the updated cluster, Ceph is not healthy"
+                fi
             fi
         fi
 
+        log "Preparing migration"
         "$DIR"/bin/kurl rook hostpath-to-block
-        "$DIR"/bin/kurl rook wait-for-health
 
-        echo "Rescaling pgs per pool"
+        log "Waiting for rook be health"
+        if ! $DIR/bin/kurl rook wait-for-health 300 ; then
+            kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph status
+            bail "Failed to verify the updated cluster, Ceph is not healthy"
+        fi
+
+        log "Rescaling pgs per pool"
         # enabling autoscaling at this point doesn't scale down the number of PGs sufficiently in my testing - it will be enabled after installing 1.4.9
         local osd_pools=
         local non_osd_pools=
         osd_pools="$(kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph osd pool ls | grep rook-ceph-store)"
         non_osd_pools="$(kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph osd pool ls | grep -v rook-ceph-store)"
         rookupgrade_10to14_scale_down_osd_pool_pg_num "$osd_pools" 16
         rookupgrade_10to14_scale_down_osd_pool_pg_num "$non_osd_pools" 32
+
         log "Waiting for pool pgs to scale down"
         rookupgrade_10to14_osd_pool_wait_for_pg_num "$osd_pools" 16
         rookupgrade_10to14_osd_pool_wait_for_pg_num "$non_osd_pools" 32
 
         logStep "Upgrading to Rook 1.1.9"
-        "$DIR"/bin/kurl rook wait-for-health
+        if ! $DIR/bin/kurl rook wait-for-health 300 ; then
+            kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph status
+            bail "Failed to verify the updated cluster, Ceph is not healthy"
+        fi
 
-        # first update rbac and other resources for 1.1
+        log "first update rbac and other resources for 1.1"
         kubectl create --save-config -f "$upgrade_files_path/upgrade-from-v1.0-create.yaml" || true # resources may already be present
         kubectl apply -f "$upgrade_files_path/upgrade-from-v1.0-apply.yaml"
 
-        # change the default osd pool size from 3 to 1
+        log "change the default osd pool size from 3 to 1"
         kubectl apply -f "$upgrade_files_path/rook-config-override.yaml"
 
         kubectl delete crd --ignore-not-found volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
         kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.1.9
         kubectl -n rook-ceph set image deploy/rook-ceph-tools rook-ceph-tools=rook/ceph:v1.1.9
-        echo "Waiting for Rook 1.1.9 to rollout throughout the cluster, this may take some time"
-        "$DIR"/bin/kurl rook wait-for-rook-version "v1.1.9"
+
+        log "Waiting for Rook 1.1.9 to rollout throughout the cluster, this may take some time"
+        if ! "$DIR"/bin/kurl rook wait-for-rook-version "v1.1.9" --timeout=600 ; then
```

Comment:
Kinda strange that we have two ways of specifying a timeout (with and without --timeout):

```
$DIR/bin/kurl rook wait-for-health 300
```
and
```
"$DIR"/bin/kurl rook wait-for-rook-version "v1.1.9" --timeout=600
```

Reply:
BTW, what is the default timeout if none is specified ?

Reply:
> Kinda strange that we have two ways of specifying a timeout (with and without --timeout):

The both methods are not the same and they exists and used in another places already.
We are not creating them in this PR. However, we in a follow up we could look to ensure that all follow the same standard. 

> BTW, what is the default timeout if none is specified ?

Then has no timeout.